lib/awful/widget/layoutbox.lua

```diff
@@ -25,7 +26,12 @@ local function update(w, screen)
     screen = get_screen(screen)
     local name = layout.getname(layout.get(screen))
     w._layoutbox_tooltip:set_text(name or "[no name]")
-    w:set_image(name and beautiful["layout_" .. name])
+
+    local exists = name and beautiful["layout_" .. name]
+        and fs.file_readable(beautiful["layout_" .. name])
+
+    w.imagebox:set_image(exists and beautiful["layout_" .. name] or nil)
+    w.textbox:set_text(exists and  "" or name)
```

Comment:
This version is fine with me, but how about the following instead (could catch more problems, but is untested):
```lua
local img = gears.surface.load_silently(beautiful["layout_" .. name], false)
if img then
    w.imagebox.image = img
    w.textbox.text = ""
else
    w.imagebox.image = nil
    w.textbox.text = name
end
```